standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1513,6 +1516,11 @@ public void commitTxn(CommitTxnRequest rqst)
         } else if (txnType.get() == TxnType.COMPACTION) {
           acquireTxnLock(stmt, false);
           commitId = getHighWaterMark(stmt);
+        } else if (txnType.get() != TxnType.READ_ONLY && isInsert(stmt, insertFilter)) {
+          stmt.executeUpdate("INSERT INTO \"WRITE_SET\" (\"WS_DATABASE\", \"WS_TABLE\", \"WS_PARTITION\", \"WS_TXNID\", \"WS_COMMIT_ID\", \"WS_OPERATION_TYPE\")" +
```

Comment:
do we have an index on TC_OPERATION_TYPE?

Reply:
no, we don't

Reply:
how would it affect the performance?

Reply:
The row count would most likely be higher in the WRITE_SET table, so selects could possibly see slower response times.